coordinator/src/node.rs

```diff
@@ -151,12 +167,18 @@ impl Node {
                     "Opening positions is disabled"
                 );
                 if let Err(e) = self.open_position(trade_params, fee_payment_hash).await {
-                    orderbook::db::orders::set_order_state(
+                    orders::set_order_state(
                         &mut connection,
                         trade_params.filled_with.order_id,
                         OrderState::Failed,
                     )?;
 
+                    matches::set_match_state(
+                        &mut connection,
+                        trade_params.filled_with.order_id,
+                        MatchState::Failed,
+                    )?;
```

Comment:
You might want to handle the error of ` orders::set_order_state(` here instead of bubling it up or update both tables atomically, otherwise it may happen that one query is run and the other one isn't.